src/api/python/CMakeLists.txt

```
@@ -129,81 +81,67 @@ add_custom_command(
-endif()
-check_cxx_compiler_flag("-Wshadow" HAVE_CXX_FLAGWshadow)
-if(HAVE_CXX_FLAGWshadow)
-  set(PY_SRC_FLAGS "${PY_SRC_FLAGS} -Wno-shadow")
-endif()
-check_cxx_compiler_flag("-Wimplicit-fallthrough" HAVE_CXX_FLAGWimplicit_fallthrough)
-if(HAVE_CXX_FLAGWimplicit_fallthrough)
-  set(PY_SRC_FLAGS "${PY_SRC_FLAGS} -Wno-implicit-fallthrough")
+copy_file_from_src(cvc5.pxi)
+copy_file_from_src(cvc5.pxd)
+copy_file_from_src(cvc5_python_base.pyx)
+copy_file_from_src(pyproject.toml)
+
+# Set include_dirs and library_dirs variables that are used in setup.cfg.in
+if (WIN32)
+  set(PYTHON_EXT "pyd")
+  set(SETUP_INCLUDE_DIRS "${PROJECT_SOURCE_DIR}/include;${CMAKE_BINARY_DIR}/include")
+  set(SETUP_LIBRARY_DIRS "${CMAKE_BINARY_DIR}/src;${CMAKE_BINARY_DIR}/src/parser")
+else()
+  set(PYTHON_EXT "so")
+  set(SETUP_INCLUDE_DIRS "${PROJECT_SOURCE_DIR}/include:${CMAKE_BINARY_DIR}/include")
+  set(SETUP_LIBRARY_DIRS "${CMAKE_BINARY_DIR}/src:${CMAKE_BINARY_DIR}/src/parser")
+  # On Linux and macOS, set rpath variable too
+  set(SETUP_RPATH "rpath=${CMAKE_BINARY_DIR}/src:${CMAKE_BINARY_DIR}/src/parser")
 endif()
-# Note: Visibility is reset to default here since otherwise the PyInit_...
-# function will not be exported correctly
-# (https://github.com/cython/cython/issues/3380).
-set_target_properties(cvc5_python_base
-                      PROPERTIES
-                      COMPILE_FLAGS "${PY_SRC_FLAGS}"
-                      CXX_VISIBILITY_PRESET default
-                      VISIBILITY_INLINES_HIDDEN 0
-                      LIBRARY_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/cvc5")
 
-python_extension_module(cvc5_python_base)
+# Set MACOS_ARCH variable that is used in setup.py.in
+if (CMAKE_CROSSCOMPILING_MACOS)
+  set(MACOS_ARCH "arm64")
+endif()
 
-# Copy the pythonic API to the right place. It does not come with its own
-# installation routine and consists only of a few files that need to go to
-# the right place.
-find_package(CVC5PythonicAPI)
+configure_file(setup.py.in setup.py)
+configure_file(setup.cfg.in setup.cfg)
+configure_file(__init__.py.in cvc5/__init__.py)
 
-set(COPIED_PYTHONIC_FILES
-  "${CMAKE_CURRENT_BINARY_DIR}/cvc5/pythonic/__init__.py"
-  "${CMAKE_CURRENT_BINARY_DIR}/cvc5/pythonic/cvc5_pythonic.py"
-  "${CMAKE_CURRENT_BINARY_DIR}/cvc5/pythonic/cvc5_pythonic_printer.py"
-)
+set(CVC5_PYTHON_BASE_LIB
+  "${CMAKE_CURRENT_BINARY_DIR}/cvc5/cvc5_python_base.${PYTHON_EXT}")
 
 add_custom_command(
   OUTPUT
-    ${COPIED_PYTHONIC_FILES}
+    ${CVC5_PYTHON_BASE_LIB}
   COMMAND
-    ${CMAKE_COMMAND} -E copy_directory
-    "${CVC5PythonicAPI_BASEDIR}/cvc5_pythonic_api"
-    "${CMAKE_CURRENT_BINARY_DIR}/cvc5/pythonic"
-  # Remove the license of the Pythonic API from the package's source files.
-  # The license file is included in the package's dist-info dir by setup()
+    # Force a new build if any dependency has changed
+    ${CMAKE_COMMAND} -E remove cvc5_python_base.cpp &&
+    ${CMAKE_COMMAND} -E remove cvc5/cvc5_python_base.${PYTHON_EXT}
```

```suggestion
    ${CMAKE_COMMAND} -E remove cvc5_python_base.cpp cvc5/cvc5_python_base.${PYTHON_EXT}
```
let's avoid the && here if possible